core/drivers/crypto/caam/utils/utils_mem.c

```diff
@@ -0,0 +1,481 @@
+		 * ligne, ensure that the size is aligned on a cache line
+		 *
+		 * Buffer address returned is moved up to a cache start offset,
+		 * hence add a cache line if the alloc_size is already aligned
+		 * otherwise the total buffer size will be too small
+		 */
+		cacheline_size = read_cacheline_size();
+		alloc_size +=
+			(alloc_size == cacheline_size) ? cacheline_size : 0;
+		alloc_size += ROUNDUP(alloc_size, cacheline_size);
+	}
+
+	ptr = malloc(alloc_size);
+	if (!ptr) {
+		MEM_TRACE("alloc Error - NULL");
+		return NULL;
+	}
+
+	if (type & MEM_TYPE_ZEROED)
+		memset(ptr, 0, alloc_size);
+
+	/* Calculate the return buffer address */
+	ret_addr = (vaddr_t)ptr + MEM_INFO_SIZE;
+	if (type & MEM_TYPE_ALIGN)
+		ret_addr = ROUNDUP(ret_addr, cacheline_size);
+
+	/*
+	 * Add the mem_info header
+	 */
+	info = OF_MEM_INFO(ret_addr);
+	info->addr = ptr;
+	info->size = size;
+	info->type = type;
+
+	MEM_TRACE("alloc returned %p -> %p", ptr, (void *)ret_addr);
+
+	return (void *)ret_addr;
+}
+
+/*
+ * Free allocated area
+ *
+ * @ptr  area to free
+ */
+static void mem_free(void *ptr)
+{
+	struct mem_info *info = NULL;
+
+	if (!ptr)
+		return;
+
+	info = OF_MEM_INFO(ptr);
+	MEM_TRACE("free %p info %p", ptr, info);
+	MEM_TRACE("free @%p - %zu bytes of type %d", info->addr, info->size,
+		  info->type);
+
+	free(info->addr);
+}
+#else
+/*
+ * Allocate an area of given size in bytes
+ *
+ * @size   Size in bytes to allocate
+ * @type   Type of area to allocate (refer to MEM_TYPE_*)
+ */
+static inline void *mem_alloc(size_t size, uint8_t type)
+{
+	void *ptr = NULL;
+	size_t alloc_size = size;
+	uint32_t cacheline_size = 0;
+
+	MEM_TRACE("alloc (normal) %zu bytes of type %d", size, type);
+
+	if (type & MEM_TYPE_ALIGN) {
+		cacheline_size = read_cacheline_size();
+		alloc_size += ROUNDUP(alloc_size, cacheline_size);
+	}
+
+	ptr = malloc(alloc_size);
+	if (!ptr) {
+		MEM_TRACE("alloc (normal) Error - NULL");
+		return NULL;
+	}
+
+	if (type & MEM_TYPE_ZEROED)
+		memset(ptr, 0, alloc_size);
+
+	MEM_TRACE("alloc (normal) returned %p", ptr);
+	return ptr;
+}
+
+/*
+ * Free allocated area
+ *
+ * @ptr  area to free
+ */
+static inline void mem_free(void *ptr)
+{
+	if (ptr) {
+		MEM_TRACE("free (normal) %p", ptr);
+		free(ptr);
+	}
+}
+#endif
+
+/*
+ * Allocate normal memory
+ *
+ * @size  size in bytes of the memory to allocate
+ */
+void *caam_alloc(size_t size)
+{
+	return mem_alloc(size, MEM_TYPE_DEFAULT);
+}
+
+/*
+ * Allocate memory aligned with a cache line
+ *
+ * @size  size in bytes of the memory to allocate
+ */
+void *caam_alloc_align(size_t size)
+{
+	return mem_alloc(size, MEM_TYPE_DEFAULT | MEM_TYPE_ALIGN);
+}
+
+/*
+ * Free allocated memory
+ *
+ * @ptr  reference to the object to free
+ */
+void caam_free(void *ptr)
+{
+	mem_free(ptr);
+}
+
+/*
+ * Allocate Job descriptor
+ *
+ * @nbentries  Number of descriptor entries
+ */
+uint32_t *caam_alloc_desc(uint8_t nbentries)
+{
+	return mem_alloc(DESC_SZBYTES(nbentries),
```

Comment:
As explain I don't want to add the overflow check for this particular function which is intern to the NXP CAAM driver. Overflow must not raised because driver has to be validated before released and this function is not called with dynamic size.

Reply:
OK